src/settings/utils/grouping.js

```diff
@@ -9,6 +9,70 @@ export const getKeyBasedArrayGroup = (array, key) => {
     return acc;
   }, {});
 };
+
+const groupTypeCapabilityByResource = (data) => Object.entries(data).reduce((acc, [type, capabilities]) => {
```

Comment:
Comments would be appreciated on these methods. Even though they are internal, the complexity is high enough to warrant some documentation.

Reply:
Added docstrings